app/interface/deps/providers.py

```
@@ -0,0 +1,54 @@
+from app.infrastructure.parsers import BarunParser, KutisParser, LmsParser
+from app.interface.schemas.common.requests import LoginRequest
+from app.interface.clients import BarunClient, KutisClient, LmsClient
+
+
+def usecase_provider_factory(
+    usecase_cls: type,
+    *,
+    dependency: Callable[..., any],
+    dep_arg: str,
+) -> Callable:
+    def _provider(dep_obj=Depends(dependency)) -> usecase_cls:
```

Ruff (0.12.2)

Missing type annotation for function argument `dep_obj`

(ANN001)